sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+# there is a stuck lvm command.
+# locking_type=0
+#
+# use_lvmetad is set to 0 in order not to show cached, old lvm metadata.
+# use_lvmetad=0
+#
+# preferred_names and filter config values are set to capture Vdsm devices.
+# preferred_names=[ '^/dev/mapper/' ]
+# filter=[ 'a|^/dev/mapper/.*|', 'r|.*|' ]
+LVM_CONFIG = """
+global {
+    locking_type=0
+    use_lvmetad=0
+}
+devices {
+    preferred_names=[ "^/dev/mapper/" ]
+    ignore_suspended_devices=1
+    write_cache_state=0
+    disable_after_error_count=3
+    filter=[ "a|^/dev/mapper/.*|", "r|.*|" ]
+}
+"""
+LVM_CONFIG = re.sub(r"\s+", " ", LVM_CONFIG).strip()
+
+
+class Vdsm(Plugin, RedHatPlugin):
+    """vdsm Plugin"""
+
+    packages = (
+        'vdsm',
+        'vdsm-client',
+    )
+
+    plugin_name = 'vdsm'
+
+    def setup(self):
+        os.environ["LVM_SYSTEM_DIR"] = "/var/run/vdsm/lvm"
+        self.add_forbidden_path('/etc/pki/vdsm/keys/*')
+        self.add_forbidden_path('/etc/pki/vdsm/libvirt-spice/*-key.*')
+        self.add_forbidden_path('/etc/pki/libvirt/private/*')
+
+        self.add_cmd_output('service vdsmd status')
+
+        self.add_copy_spec(['/tmp/vds_installer*',
+                            '/tmp/vds_bootstrap*',
+                            '/etc/vdsm/*'])
+
+        limit = self.get_option('log_size')
+
+        if limit is not None:
+            self.add_copy_spec('/var/log/vdsm/*', sizelimit=limit)
+        else:
+            self.add_copy_spec('/var/log/vdsm/*')
+
+        self._add_vdsm_run_dir()
+        self.add_copy_spec(['/usr/libexec/vdsm/hooks',
+                            '/var/lib/vdsm',
+                            '/var/log/ovirt.log',
+                            '/var/log/sanlock.log'])
+
+        res = self.call_ext_prog('/usr/bin/pgrep qemu-kvm')
+        if res['status'] == 0:
+            for line in res['output']:
+                pid = line.strip()
+                self.add_copy_spec(["/proc/%s/cmdline" % pid,
+                                    "/proc/%s/status" % pid,
+                                    "/proc/%s/mountstats" % pid])
+        self.add_cmd_output(["/bin/ls -ldZ /etc/vdsm",
+                            "/bin/su vdsm -s /bin/sh -c\
+                             '/usr/bin/tree -l /rhev/data-center'",
+                             "/bin/su vdsm -s /bin/sh -c\
+                              '/bin/ls -lR /rhev/data-center'"])
+        self.add_cmd_output("/sbin/lvm vgs -v -o +tags --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_vgs_v_o_tags")
+        self.add_cmd_output("/sbin/lvm lvs -v -o +tags --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_lvs_v_o_tags")
+        self.add_cmd_output("/sbin/lvm pvs -v -o +all --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_pvs_v_o_all")
```

Comment:
I would be a lot happier if we could find a way to include this in the standard `lvm` plugin. Also, please do not use `suggest_filename` unless absolutely necessary: it is intended only to allow overriding an automatically generated file name when it would either be excessively long for readability purposes, or would exceed a path length limit and so lose uniqueness.

Reply:
I was asked in the previous review to use `suggest_filename` because the generated file name is too long.

Reply:
By whom? I can't find it anywhere in the history for this one - it is something a lot of new contributors do, because they think it is more "friendly" - we reserve its use for *very* long command lines - things that could even blow the filename length limit for example.

This includes RPM commands with many dozens or hundreds of packages on the command line, as well as the `modinfo` calls that run for every kernel module on the system.

The reason it's harmful to overuse it is that it prevents the actual command run from being mechanically recovered from the file name, and it prevents something trying to comprehend the archive from doing the reverse: determining the output file name from the command run.

Reply:
If it was me then I apologise - but I'd be surprised as these command lines are really not that long at all...

Reply:
No problem, I will remove it.